### PR TITLE
Fixes no space views showing up for later logged entities

### LINKED ISSUE #3413: Not all plots show up correctly via heuristics
**Describe the bug**
Only 2 plots show up.

**To Reproduce**
Clear the blueprint:
```
rm /home/jleibs/.local/share/rerun/blueprints/rerun_example_plot.blueprint
```

Run the plots example:
```
python examples/python/plots/main.py
```

Clicking the repopulate button causes all 4 plots to show up.

**Expected behavior**
All 4 plots should show up.

**Rerun version**
`main`
Does not repro on 0.8.2

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* Fixes #3413

Currently, if any significant user change occurs to the blueprint we no longer add any new space views. For this we relied on detecting arbitrary changes to the tile tree. This causes a lot of false positive  change detections. Setting `has_been_user_edited` is more explicit now again. Major drawback is that this easily doesn't detect all possible changes. For instance, with the fix as is it doesn't detect manual relayouting as a change - which arguably isn't enough of a change to warrant not adding new space views. Eventually, we'll have to put the long planned ui for auto-adding space views in place, but this fix will have to do for now

Tested with
```
rm -rf ~/Library/Application\ Support/rerun/blueprints
python ./examples/python/plots/main.py      
```
image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3454) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/3454)
- [Docs preview](https://rerun.io/preview/fae91fd7f14a8811893d4c35fdd9ebcf8dc3dc09/docs) 
- [Examples preview](https://rerun.io/preview/fae91fd7f14a8811893d4c35fdd9ebcf8dc3dc09/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

File: crates/re_viewport/src/viewport.rs [hunk 1]
```diff
@@ -146,19 +146,14 @@ impl<'a, 'b> Viewport<'a, 'b> {
...
             ui.spacing_mut().item_spacing.x = re_ui::ReUi::view_padding();
 
             re_tracing::profile_scope!("tree.ui");
-            let tree_before = tree.clone();
             tree.ui(&mut tab_viewer, ui);
-
-            // Detect if the user has moved a tab or similar:
-            if !blueprint.has_been_user_edited && *tree != tree_before {
-                blueprint.has_been_user_edited = true;
-            }
```
Comment (emilk) [inline crates/re_viewport/src/viewport.rs:161]: This was an essential fix in https://github.com/rerun-io/rerun/pull/3295 for https://github.com/rerun-io/rerun/issues/2880

Comment (emilk) [reply crates/re_viewport/src/viewport.rs:161]: Maybe a better fix is to have two bools - one for deciding wether to auto-add new stuff, and one for deciding wether to do full re-layout

Comment (Wumpf) [reply crates/re_viewport/src/viewport.rs:161]: let's try that!

**Comment by Wumpf** [comment]
turns out we already had `auto_space_views` which was strongly coupled to `has_been_user_edited`. Fixed that, changed `has_been_user_edited` -> `auto_layout` and adjusted all the usages :)

Made sure that both #2880 and #3413 is fixed after that 👍

File: crates/re_viewport/src/viewport_blueprint.rs [hunk 2]
```diff
@@ -47,8 +47,10 @@ pub struct ViewportBlueprint<'a> {
     /// Show one tab as maximized?
     pub maximized: Option<SpaceViewId>,
 
-    /// Set to `true` the first time the user messes around with the viewport blueprint.
-    pub has_been_user_edited: bool,
+    /// Whether the viewport layout id determined automatically.
```
Comment (emilk) [inline crates/re_viewport/src/viewport_blueprint.rs:50]:
```suggestion
    /// Whether the viewport layout is determined automatically.
```

File: crates/re_viewport/src/viewport_blueprint.rs [hunk 3]
```diff
@@ -47,8 +47,10 @@ pub struct ViewportBlueprint<'a> {
...
+    ///
+    /// Set to `false` the first time the user messes around with the viewport blueprint.
+    pub auto_layout: bool,
 
     /// Whether or not space views should be created automatically.
     pub auto_space_views: bool,
```
Comment (emilk) [inline crates/re_viewport/src/viewport_blueprint.rs:56]: nice that these bools have the same "sign" now

**Review by emilk — APPROVED** [review]

**Pull request merged**